src/Test/WordPress/ProxyTest.php

```diff
@@ -2,78 +2,143 @@
+        $mockWPVerifyNonce->expects($this->once())->willReturn(true);
+        $this->mockRequestRouter->expects($this->once())->method('route');
+        $mockWPDie = $this->getFunctionMock('CF\WordPress', 'wp_die');
+        $this->proxy->run();
+    }
+
+    public function testIsCloudFlareCSRFTokenValidReturnsTrueForGet()
+    {
+        $this->assertTrue($this->proxy->isCloudFlareCSRFTokenValid('GET', null));
+    }
+
+    public function testCreateRequestGETProxyClient()
+    {
+        $url = 'testproxyurl.com';
+        $proxyURL = 'proxyurl.com';
+        $_SERVER['REQUEST_METHOD'] = 'GET';
+        $_GET['proxyURL'] = $url;
+        $_GET['proxyURLType'] = 'CLIENT';
+        $jsonBody = json_encode(array(
+            'proxyURL' => $proxyURL,
+        ));
+
+        $mockFileGetContents = $this->getFunctionMock('CF\WordPress', 'file_get_contents');
+        $mockFileGetContents->expects($this->any())->willReturn($jsonBody);
+
+        $request = $this->proxy->createRequest();
 
+        $this->assertFalse(isset($request->getParameters()['proxyURL']));
+        $this->assertFalse(isset($request->getParameters()['proxyURLType']));
+        $this->assertFalse(isset($request->getBody()['proxyURL']));
+        $this->assertEquals(\CF\API\Client::ENDPOINT.$url, $request->getUrl());
+    }
+
+    public function testCreateRequestGETProxyPlugin()
+    {
+        $url = 'testproxyurl.com';
+        $proxyURL = 'proxyurl.com';
+        $_SERVER['REQUEST_METHOD'] = 'GET';
+        $_GET['proxyURL'] = $url;
+        $_GET['proxyURLType'] = 'PLUGIN';
+        $jsonBody = json_encode(array(
+            'proxyURL' => $proxyURL,
+        ));
+
+        $mockFileGetContents = $this->getFunctionMock('CF\WordPress', 'file_get_contents');
+        $mockFileGetContents->expects($this->any())->willReturn($jsonBody);
+
+        $request = $this->proxy->createRequest();
+
+        $this->assertFalse(isset($request->getParameters()['proxyURL']));
+        $this->assertFalse(isset($request->getParameters()['proxyURLType']));
+        $this->assertFalse(isset($request->getBody()['proxyURL']));
+        $this->assertEquals(\CF\API\Plugin::ENDPOINT.$url, $request->getUrl());
+    }
+
+    public function testCreateRequestNonGET()
+    {
+        $proxyURL = 'proxyurl.com';
+        $_SERVER['REQUEST_METHOD'] = 'POST';
+        $jsonBody = json_encode(array(
+            'proxyURL' => $proxyURL,
+        ));
+
+        $mockFileGetContents = $this->getFunctionMock('CF\WordPress', 'file_get_contents');
+        $mockFileGetContents->expects($this->any())->willReturn($jsonBody);
+
+        $request = $this->proxy->createRequest();
+
+        $this->assertFalse(isset($request->getParameters()['proxyURL']));
+        $this->assertFalse(isset($request->getParameters()['proxyURLType']));
+        $this->assertFalse(isset($request->getBody()['proxyURL']));
+        // $this->assertEquals($proxyURL, $request->getUrl());
```

Comment:
Your approach is fine, I just don't like partial mocks.  

The constructor approach would have worked, when the class was instantiated it would grab the JSON, then it would be available when `run()` was called.  I'm not sure what you mean by restricting your ability to move the constructor.